src/pages/user/Profile.js

```diff
@@ -1,18 +1,205 @@
+import PropTypes from 'prop-types';
 import { Helmet } from 'react-helmet';
-import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
 import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { Link } from 'react-router-dom';
+import { Form as FinalForm, Field } from 'react-final-form';
 import {
   Container,
   Row,
   Col,
   Card,
   CardGroup,
   ButtonGroup,
-  Button
+  Button,
+  Form,
+  InputGroup
 } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { actions as appActions } from 'reducers/application';
+import { actions as profileActions } from 'reducers/profile';
+import { isLoggedIn } from 'selectors/application';
+import { getCurrentUser, getUserProfile } from 'selectors/profile';
+
+export class Profile extends Component {
+  static propTypes = {
+    match: PropTypes.object.isRequired,
+    actions: PropTypes.object.isRequired,
+    collection: PropTypes.any,
+    currentUser: PropTypes.bool,
+    loggedIn: PropTypes.bool.isRequired,
+    editting: PropTypes.bool.isRequired
+  };
+  constructor(props) {
+    super(props);
+    const {
+      actions,
+      match: { params, path }
+    } = this.props;
+
+    this.state = { editting: false };
+
+    if (path === '/user/profile' || path === '/user') {
+      actions.requestCurrentUserProfile();
+    }
+    if (path === '/user/:userName') {
+      actions.requestProfile({ name: params.userName });
+    }
+
+    this.handleEditor = this.handleEditor.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  get Gravatar() {
```

Comment:
What's your rationale for capitalizing these getters?

Reply:
Originally I had `collection.gravatar` and `collection.website`, so I capped them here to prevent a conflict. They aren't in `collection` anymore so I'll lc them.